utils/blkdiscard_test.go

```diff
@@ -27,3 +29,71 @@ func Test_blkdiscard(t *testing.T) {
 	// fake-block-device isn't a blockdevice that supports TRIM so we expect an error
 	assert.Error(t, err)
 }
+
+// Test blkdiscard using a file mounted as a loopback device. This test requires
+// root privileges, and is only run if the ENABLE_PRIV_TESTS environment variable
+// is set to true.
+func Test_blkdiscard_loopback(t *testing.T) {
+	if os.Getenv("ENABLE_PRIV_TESTS") != "true" {
+		t.Skip("Skipping privileged test blkdiscard_loopback")
+	}
+
+	tempDir := t.TempDir()
+
+	// Create a 10 MB temporary file
+	fileSize := int64(10 * 1024 * 1024)
+	fileName := tempDir + "testfile"
+	knownOdOutput := `0000000 000000 000000 000000 000000 000000 000000 000000 000000
+*
+50000000
+`
+	f, err := os.Create(fileName)
+	assert.NoError(t, err)
+
+	// Fill the file with 1's
+	_, err = f.WriteAt(bytes.Repeat([]byte{1}, int(fileSize)), 0)
+	f.Close()
+	assert.NoError(t, err)
+	t.Cleanup(func() { os.Remove(fileName) })
+
+	t.Log("hi")
+	// Create a loopback device using losetup
+	losetupOutput, err := exec.Command("losetup", "--find").Output()
+	assert.NoError(t, err)
+	loopbackDevice := string(bytes.TrimSpace(losetupOutput))
+	t.Logf("DEBUG: loopbackDevice will be: %s", loopbackDevice)
+	_, err = exec.Command("losetup", "--find", fileName).Output()
```

Comment:
why run losetup --find twice? This is liable to toctou race and loopbackDevice being incorrect. Instead, you can run `losetup --associated filename` and it returns all the loopback devices attached to this file (hopefully only one :) )

Reply:
It was being weird on my machine. I'll try it your way.

Reply:
cleaned up.